tests/e2e_decorrelation/test_nested_subqueries.py

```
@@ -0,0 +1,704 @@
+
+        parser = Parser()
+        binder = Binder(catalog)
+        decorrelator = Decorrelator()
+
+        logical_plan = parser.parse(sql)
+        bound_plan = binder.bind(logical_plan)
+
+        # Skip-level correlation (innermost subquery referencing the
+        # outermost query) is documented future work; the engine must
+        # refuse loudly rather than rewrite it incorrectly.
+        with pytest.raises(DecorrelationError, match="u.id"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)